x-pack/plugins/ml/public/components/rule_editor/save_rule_modal.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import PropTypes from 'prop-types';
+import React, {
+  Component,
+} from 'react';
+
+import {
+  EuiButton,
+  EuiConfirmModal,
+  EuiOverlayMask,
+  EUI_MODAL_CONFIRM_BUTTON,
+} from '@elastic/eui';
+
+
+/*
+ * React modal for validating and saving edits to a rule.
```

Comment:
Is this modal complete?
It looks like a WIP?

Reply:
Well spotted. I'll remove the `save_rule_modal.js` file as it isn't needed. The checks it was going to do on save are now done inline, and the Save button is disabled if the edits are not valid e.g. if neither a condition or scope has been added.